graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,273 @@
+import numpy as np
+from ..match import GraphMatch as GMP
+from sklearn.base import BaseEstimator
+import itertools
+
+
+class VNviaSGM(BaseEstimator):
+    """
+    This class implements vertex nomination via seeded graph matching
+
+    Parameters
+    ----------
+    G_1: 2d-array, square
+        Adjacency matrix of `G_1`, the graph where voi is known
+
+    G_2: 2d-array, square
+        Adjacency matrix of `G_2`, the graph where voi is not known
+
+    h: int
+        distance between voi used to create induced subgraph on `G_1`
+
+    ell: int
+        distance from seeds to other verticies to create induced subgraphs on `G_1`
+
+    R: int
+        Number of restarts for soft seeded graph matching algorithm
+
+    Attributes
+    ----------
+    n_seeds_used: int
+        Number of seeds passed in `seedsA` that occured in the induced subgraph about `voi`
+
+    nomination_list: list
+        List of 2 tuples of the format nomination_list = [(j, p_val)] where p_val is the
+        probability that the voi matches to node j in graph B (sorted by descending
+        probability)
```

Comment:
I would think that `nomination_list` and `p_val` should be represented and returned from predict as a numpy array for consistency and efficient use with other parts of graspologic. Could be ``Tuple[ndarray, ndarray]``.